service/src/main/java/com/networknt/service/SingletonServiceFactory.java

```diff
@@ -77,8 +77,14 @@ private static Object handleSingleImpl(List<String> interfaceClasses, List<Objec
         List<Object> items = new ArrayList<>();
         if (it.hasNext()) {
             Map.Entry<String, Map<String, Object>> pair = (Map.Entry) it.next();
+
+            // check whether intent is to generate implementation from factory method
             String key = pair.getKey();
-            Class implClass = Class.forName(key);
+            String initClassName = key.contains("::") ? key.substring(0, key.indexOf("::")) : key;
+            String initMethodName = key.contains("::") ? key.substring(key.indexOf("::") + 2) : null;
```

Comment:
Can the "::" be moved to a constant, and the + 2 be referenced as the length of the constant?